code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -105,7 +105,7 @@ GLOBAL_LIST_INIT(blacklisted_borg_hats, typecacheof(list( //Hats that don't real
 		return
 
 	if(istype(W, /obj/item/encryptionkey) && opened)
-		if(radio)//sanityyyyyy
+		if(radio && !shell)//removed ability to add key to AI shell, this is done through the AI core.
```

Comment:
You should add a unique feedback message for shells, rather than `Unable to locate a radio!`. Like `No radio slot!`. Also balloon alert it

Reply:
adjusted to your request